Add stress tests for gRPC, HTTP, and TCP liveness probesAdd probe stress tests

KasimVali2207 opened the pull request:
This PR adds e2e stress tests for gRPC, HTTP, and TCP liveness probes to address issue #115782.

Each test creates 50 containers with liveness probes running every 1 second, waits 2 minutes, and validates that no containers have restarted unexpectedly.

The tests use `framework.WithSerial()` to ensure serial execution and prevent resource contention from running multiple high-load stress tests simultaneously.

This implementation:
- Serves as a replacement for the skipped unit test from #115329
- Helps verify the bug fix from #89898

KasimVali2207 committed:
Add stress tests for gRPC, HTTP, and TCP liveness probes

This commit adds e2e stress tests to verify that liveness probes do not cause
unexpected container restarts under load. The tests create many containers
(50 per test) with liveness probes configured to run every 1 second.

Three test cases are included:
- HTTP liveness probe stress test
- TCP liveness probe stress test
- gRPC liveness probe stress test

Each test waits for all containers to be running, observes probe behavior for
2 minutes, and validates that no containers have restarted unexpectedly.

These tests address the bug fix from issue kubernetes#89898 and serve as a
replacement for the skipped unit test from PR kubernetes#115329.

KasimVali2207 commented on test/e2e_node/probe_stress_test.go:
```diff
@@ -0,0 +1,232 @@
+	probeStressWaitTime      = 2 * time.Minute
+)
+
+var _ = SIGDescribe("Probe Stress", framework.WithSerial(), func() {
```
The tests are marked with `framework.WithSerial()` which ensures they run serially. This is intentional because stress tests create high load on the system, and running them in parallel could lead to resource contention and unreliable test results.

bart0sh replied:
> This PR removes redundant framework.WithSerial() decorators from probe stress tests as they were causing test failures. All tests are now passing and ready for review.

That comment sounds kinda contradictional to what you've written above. If serial tests were intentional they shouldn't be removed, I guess. Instead test failures should be investigated and fixed.

bart0sh commented on test/e2e_node/probe_stress_test.go:
```diff
@@ -0,0 +1,232 @@
+
+var _ = SIGDescribe("Probe Stress", framework.WithSerial(), func() {
+	f := framework.NewDefaultFramework("probe-stress")
+	f.NamespacePodSecurityLevel = admissionapi.LevelPrivileged
```
Can you add a comment explaining why `LevelPrivileged` is used here?

bart0sh commented on test/e2e_node/probe_stress_test.go:
```diff
@@ -0,0 +1,232 @@
+			RestartPolicy: v1.RestartPolicyNever,
+		},
+	}
+}
```
@KasimVali2207 Have you pushed your changes to github or just committed them locally :) ? I don't see anything changed since I looked last time.

KasimVali2207 committed:
Address review feedback: use gomega.Consistently and refactor pod creation

- Replaced time.Sleep with gomega.Consistently to ensure reliable failure detection if a restart occurs during the wait
- Refactored createPodWith*Probes functions into a single helper to remove code duplication

Signed-off-by: KasimVali2207 <kasimvali2207@gmail.com>

bart0sh commented on test/e2e_node/probe_stress_test.go:
```diff
@@ -0,0 +1,187 @@
+	containers := make([]v1.Container, numContainers)
+
+	for i := 0; i < numContainers; i++ {
+		probe, ports, args := generator(i)
```
What's the purpose of `generator` here? Would it work if `createProbeStressPod` take 3 more args instead?

SergeyKanzhelev reviewed (approved):
/lgtm
/approve
